common/task/sequentialTaskProcessor.go

```diff
@@ -90,105 +81,85 @@ func (t *sequentialTaskProcessorImpl) Stop() {
 }
 
 func (t *sequentialTaskProcessorImpl) Submit(task SequentialTask) error {
-	hashCode := int(task.HashCode()) % t.coroutineSize
-	taskQueue := t.coroutineTaskQueues[hashCode]
+
+	taskqueue := task.NewTaskQueue()
```

Comment:
changing the NewTaskQueue interface method to input factory function which can generate corresponding task queue for each type of task

Reply:
it's a bit weird to ask a task to return a taskQ on which it should be enqueued for processing by a workerPool owned by the calling thread. Can you instead consider having your tasks implement one method - PartitionKey() Key  or GroupingKey() Key - where you can alias Key to WorkflowIdentifier for now - that way, you can create the queue here instead of creating the queue from within the task